Update dotnet sln (#15694)
dotnet/docs: docs/core/tools/dotnet-sln.md

## Name

> `dotnet sln` - Options for listing or modifying the projects in a .NET Core solution file.

Reviewer: ```suggestion
`dotnet sln` - Lists or modifies the projects in a .NET Core solution file.
```

<!-- todo: uncomment when all CLI commands are reviewed
[!INCLUDE [topic-appliesto-net-core-all](../../../includes/topic-appliesto-net-core-all.md)]
-->
## Name

> `dotnet sln` - Options for the projects in a .NET Core solution file.

Reviewer 2: I think the previous wording was actually better here.

Reviewer 3: Wondering if we should break this down into multiple pages like we did with dotnet tool

- **`SOLUTION_FILE`**

>   The solution file to use. The command searches the current directory for one if it's left unspecified, failing if there are multiple solution files.

Reviewer: ```suggestion
  The solution file to use. If the argument is omitted, the command searches the current directory. If it finds no solution file or multiple solution files, the command fails.
```

## Options

- **`-h|--help`**

  Prints options for the command.

## Commands

### `list`

Lists all projects in a solution file.

#### Synopsis

```dotnetcli
dotnet sln list [-h|--help]
```
  
#### Arguments

- **`SOLUTION_FILE`**

>   The solution file to use. The command searches the current directory for one if it's left unspecified, failing if there are multiple solution files.

Reviewer: In the H4 Arguments sections I would omit the description for SOLUTION_FILE since that's covered at the top in the H2 Arguments section. If we keep the H4 descriptions we should make sure they're all the same.

Author: I'd prefer to keep them, since I can easily imagine people skipping down to what they think is relevant, then getting confused. I made this one match the one above, though.

- **`SOLUTION_FILE`**

  The solution file to use. The command searches the current directory for one if it's left unspecified, failing if there are multiple solution files.

## Options

- **`-h|--help`**

  Prints options for the command.

## Commands

### `list`

Lists all projects in a solution file.

#### Synopsis

```dotnetcli
dotnet sln list [-h|--help]
```
  
#### Arguments

- **`SOLUTION_FILE`**

  The solution file to use. The command searches the current directory for one if it's left unspecified, failing if there are multiple solution files.

#### Options

- **`-h|--help`**

  Prints out options for the command.
  
### `add`

Adds one or more projects to the solution file.

#### Synopsis

```dotnetcli

> dotnet sln [<SOLUTION_FILE>] add [--in-root] [-s|--solution-folder <SOLUTION_FOLDER>] <PROJECT_PATH> [<PROJECT_PATH>...]

Reviewer 3: We don't usually show the argument in the synopsis. Instead, we add that where the option is shown.

```suggestion
dotnet sln [<SOLUTION_FILE>] add [--in-root] [-s|--solution-folder] <PROJECT_PATH> [<PROJECT_PATH>...]
```

Reviewer 3: I just saw another PR where a customer was trying to add the same info. Do you two think that we should do that across the board? I closed the PR for now so we could discuss that.

Author: If switches are well-named (as in `--solution-folder`), I don't think it's necessary, but if `-s` were the only way to express that, it would be better to include `<SOLUTION_FOLDER>`. Since consistency is good, I would say that if there is _one_ case in which we can't have a well-named switch, we should do it across the board. Otherwise, it's fine to take it out.

I'll take it out for now.

Reviewer 2: To me it makes sense to show the argument in the synopsis in all cases.  Otherwise you can't tell the difference between a switch like `--in-root` which doesn't accept an argument and one like `--solution-folder` which requires one.

dotnet sln add [-h|--help]
```

#### Arguments

- **`SOLUTION_FILE`**

>   The solution file to use. The command searches the current directory for one if it's left unspecified, failing if there are multiple solution files.

Reviewer 3: It's part of our style guide to start the sentence with the condition first.

- **`PROJECT_PATH`**

  The path to the project or projects to add to the solution. Unix/Linux shell [globbing pattern](https://en.wikipedia.org/wiki/Glob_(programming)) expansions are processed correctly by the `dotnet sln` command.

#### Options

- **`-h|--help`**

>   Prints out a short list of options for the command.

Reviewer 3: this is description for all --help options across CLI docs. If this is preferred version, we'd have to change across all docs.

- **`--in-root`**

> 

Reviewer 3: line 100 would be the place to add the option argument for -s

Author: I'm not following. -s is on line 100.

- **`SOLUTION_FILE`**

>   The solution file to use. If not specified, the command searches the current directory for one. It fails if there are multiple solution files in the directory.

Reviewer 2: I'd suggest combining the last two sentences to make clear that it won't fail if there are multiple solution files but one is specified on the command line.

- **`PROJECT_PATH`**

>   The path to the project to add to the solution. Add multiple projects by adding one after the other separated by spaces. (See the fourth example.) Unix/Linux shell [globbing pattern](https://en.wikipedia.org/wiki/Glob_(programming)) expansions are processed correctly by the `dotnet sln` command.

Reviewer 2: It might be best not to add "(See the fourth example.)".  It's likely to become incorrect if we add, remove, or reorder the examples.

#### Synopsis

```dotnetcli
dotnet sln [<SOLUTION_FILE>] remove <PROJECT_PATH> [<PROJECT_PATH>...]
dotnet sln [<SOLUTION_FILE>] remove [-h|--help]
```

#### Arguments

- **`SOLUTION_FILE`**

>   The solution file to use. The command searches the current directory for one if it's left unspecified, failing if there are multiple solution files.

Reviewer 3: again, starting with the condition is part of our style and voice

- **`PROJECT_PATH`**

  The path to the project or projects to add to the solution. Unix/Linux shell [globbing pattern](https://en.wikipedia.org/wiki/Glob_(programming)) expansions are processed correctly by the `dotnet sln` command.

#### Options

- **`-h|--help`**

>   Prints out configurable options for the command.

Reviewer 3: This looks different from the previous description for help.

Author: I included two versions to see which you like better, but I don't like the current one. It's bad style to include a word in its definition, and this usage for "help" is awkward in English in my opinion, so I would prefer to use one of my versions. Do you like either?

Reviewer 3: Would we change in the CLI too? It's not exactly the same but it says help too.

I don't think that --help is only about options so I think both changes are a bit misleading.

Author: I would.

That's true. How about one of:
"Emits a description of how to use dotnet sln list"
"Prints out a description of how to use dotnet sln list"
"Describes how to use dotnet sln list"
possibly replacing "dotnet sln list" with "this command"?

Reviewer: I vote for "Prints out a description of how to use the command."  Whatever we end up with here we can propagate to the other CLI docs later.

- **`SOLUTION_FILE`**

  The solution file to use. If not specified, the command searches the current directory for one. It fails if there are multiple solution files in the directory.

- **`PROJECT_PATH`**

>   The path to the project to remove from the solution. Remove multiple projects by adding one after the other separated by spaces. (See the fifth example.) Unix/Linux shell [globbing pattern](https://en.wikipedia.org/wiki/Glob_(programming)) expansions are processed correctly by the `dotnet sln` command.

Reviewer 2: Same here, probably best not to specify example numbers.
